openedx_external_enrollments/external_enrollments/pathstream_external_enrollment.py

```diff
@@ -0,0 +1,201 @@
+"""PathstreamExternalEnrollment class file."""
+import logging
+import os
+from datetime import datetime
+
+import boto3
+from botocore.exceptions import ClientError
+from django.conf import settings
+
+from openedx_external_enrollments.external_enrollments.base_external_enrollment import BaseExternalEnrollment
+from openedx_external_enrollments.models import ExternalEnrollment
+
+LOG = logging.getLogger(__name__)
+COMPLETED = True
+UNCOMPLETED = False
+
+
+class S3NotInitialized(BaseException):
+    """Exception when _init_s3 has not been initialized before calling _upload_file or _download_file """
+
+
+class PathstreamExternalEnrollment(BaseExternalEnrollment):
+    """
+    PathstreamExternalEnrollment class.
+    """
+    def __init__(self):
+        self.client = None
```

Comment:
Yes according to the logic of the external enrollment, If you check out _download_file and _upload_file, you will find out that in that case that `self.client = None` , it will raise an error  `S3NotInitialized,` this is due to the fact that before attempting to either `_download_file` or `_upload_file`,  `_init_s3` must have been called before.

It also makes no sense to call  `_init_s3` within '__init__' since the S3 connection only will be used when the periodic task executes. @ivanvgh  Is that clear?